python_modules/dagster/dagster/core/host_representation/external.py

```diff
@@ -488,15 +488,31 @@ def name(self):
 
     @property
     def pipeline_name(self):
-        return self._external_sensor_data.pipeline_name
+        target = self._get_single_target()
+        return target.pipeline_name if target else None
 
     @property
-    def solid_selection(self):
-        return self._external_sensor_data.solid_selection
+    def mode(self):
+        target = self._get_single_target()
+        return target.mode if target else None
 
     @property
-    def mode(self):
-        return self._external_sensor_data.mode
+    def solid_selection(self):
+        target = self._get_single_target()
+        return target.solid_selection if target else None
+
+    def _get_single_target(self):
+        if self._external_sensor_data.target_dict:
+            check.invariant(len(self._external_sensor_data.target_dict.values()) == 1)
```

Comment:
is invariant violation ok here? at least want an explanatory message

Reply:
removed, especially since we still need to resolve how we display these sensors in cragit